src/test/java/educationalproject/programmingstuff/tests_units/UserControllerTest.java

```diff
@@ -1,39 +1,43 @@
 package educationalproject.programmingstuff.tests_units;
 
-import educationalproject.programmingstuff.TestDataFactory;
-import educationalproject.programmingstuff.controller.UserController;
+import com.fasterxml.jackson.databind.ObjectMapper;
 import educationalproject.programmingstuff.servicies.UserServiceImpl;
 import educationalproject.programmingstuff.servicies.dto.UserCreateRequestDto;
 import org.junit.jupiter.api.Test;
-import org.junit.jupiter.api.extension.ExtendWith;
 import org.mockito.Mockito;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureWebMvc;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
 import org.springframework.boot.test.mock.mockito.MockBean;
-import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
 
 import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.Mockito.times;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
 
-//@WebMvcTest doesn't work fo me and for dude from stackoverflow as well (look down of the page): https://stackoverflow.com/questions/48078044/webmvctest-fails-with-java-lang-illegalstateexception-failed-to-load-applicati
-@SpringBootTest
-@ExtendWith(SpringExtension.class)
+@WebMvcTest
+@AutoConfigureMockMvc
+@AutoConfigureWebMvc
```

Comment:
@WebMvc(UserController.class) should be enough
see https://spring.io/guides/gs/testing-web/

Reply:
Done. Read the article